prow/config/config.go

```diff
@@ -622,6 +624,9 @@ func (pc *ProwConfig) mergeProwJobDefault(repo, cluster string, jobDefault *prow
 	if merged == nil {
 		merged = &prowapi.ProwJobDefault{}
 	}
+	if merged.TenantID == "" {
```

Comment:
I think this would work. but imo this should happen earlier, more specifically prior to calling `mergeProwJobDefault`. this is mimicking the behavior of inserting one entry on top of `ProwJobDefaultEntry` as:
```
- config:
    tenant_id: <DEFAULT_ID>
```

Reply:
If we make changes to the ProwJobDefault before calling `mergeProwJobDefault` then those changes will always override any other defaults. I am also not sure I understand the problem with the current implementation.

Reply:
The idea I had in mind was adding 
```
- config:
    tenant_id: <DEFAULT_ID>
```
as the first entry in `pc.ProwJobDefaultEntries`, which won't override anything else, but has its own problem. I think your solution is actually better